### DISCUSSION TITLE
x-if seems buggy / sticky? [DISREGARD: Double post, cant find way to delete]

### DISCUSSION THREAD

**Discussion by tbjolset** [alpinejs/alpine#2541] [7. Bugs (archived)]
Try the example below: Initially, user is null, and the x-if leaves it alone. Tap log in, x-if now shows the logged in user. Then tap logout, user is now null again. The x-if correctly doesnt try to show the user any more, but still tries to evaluat user.name for some reason, causing console errors etc. In practise, this makes it hard/impossible for me to do component based design.

```
<html lang="en">
<head>
  <script src="//unpkg.com/alpinejs" defer></script>
</head>

<body x-data="{ user: null }">
  <template
    x-if="user !== null"
  >
    <div>
      <div x-text="'user: ' + user.name"></div>
      <button @click="user = null">Log out</button>
    </div>
  </template>
  <template x-if="!user">
    <button @click="user = { name: 'lebowski' }">Log in</button>
  </template>

</body>

</html>
```

(sorry if this somehow got double posted, i tried to submit first but couldnt find my original question when searching)

**Comment by SimoTod** [comment]
This project uses discussions (in case you were trying to find the first question in the issue tab).
